### PR TITLE
Updates to index.ts and implementation for SalesOrderDetails FE

### PR DESCRIPTION
**Pull request by zixuan97**
feat: Update sales order, can add/remove items into SalesOrderItems[]
fix: Update of index.ts to sync with BE

### CODE REVIEW

**Comment by zixuan97** [comment]
They'll need the types.ts updated file, I'll refactor after I'm done implementing the routes to connect to BE

File: src/pages/sales/SalesOrderDetails.tsx [hunk 1]
```diff
@@ -68,56 +76,129 @@ const steps = [
...
+              default: {
+                break;
+              }
+            }
+            setLoading(false);
+          } else {
+            setAlert({
+              severity: 'error',
+              message:
+                'Sales Order does nto exist. You will be redirected back to the Sales Order Overview page.'
```
Comment (danielonges) [inline src/pages/sales/SalesOrderDetails.tsx:194]: typo

File: src/pages/sales/SalesOrderDetails.tsx [hunk 2]
```diff
@@ -68,56 +76,129 @@ const steps = [
...
+            setSalesOrder(salesOrder);
+            setEditSalesOrderItems([...salesOrder.salesOrderItems]);
+            setAvailProducts(
+              products.filter((product) => {
+                return salesOrder?.salesOrderItems.find(
+                  (orderItem) => orderItem.productName !== product.name
+                );
+              })
+            );
+            switch (salesOrder.orderStatus) {
```
Comment (danielonges) [inline src/pages/sales/SalesOrderDetails.tsx:159]: this whole switch statement can be refactored by including the OrderStatus as a value in each object within the steps array, then finding the index of the object that matches the value

`const steps = [{
    label: 'Order Placed',
    value: OrderStatus.CREATED,
    icon: <ReceiptLongRounded sx={{ fontSize: 35 }} />,
    nextAction: 'Confirm Payment'
  },...`

then afterwards,
```suggestion
            setActiveStep(steps.findIndex(step => step.value === salesOrder.orderStatus)
```

File: src/pages/sales/SalesOrderDetails.tsx [hunk 3]
```diff
@@ -68,56 +76,129 @@ const steps = [
...
         }
-      }
-      setLoading(false);
-    }
-  }, [salesOrders]);
+      );
+  }, [id, navigate, products]);
 
   const nextStep = () => {
     switch (salesOrder?.orderStatus) {
```
Comment (danielonges) [inline src/pages/sales/SalesOrderDetails.tsx:204]: this switch statement also can be refactored
```suggestion
    if (activeStep < steps.length - 1) {
      setActiveStep(prev => prev + 1);
      const newStatus = steps[activeStep + 1].value;
      if (newStatus === OrderStatus.PREPARED) {
        setSalesOrder(order => order && {...order, orderStatus: newStatus, salesOrderItems: editSalesOrderItems });
        updateSalesOrder();
      } else {
        setSalesOrder(order => order && {...order, orderStatus: newStatus });
      }
    }
```

File: src/services/salesService.ts [hunk 4]
```diff
@@ -2,14 +2,18 @@ import axios from 'axios';
...
     .get(`${apiRoot}/sales/all`)
     .then((res) => res.data);
 };
+
+export const getSalesOrderDetailsSvc = (id: string): Promise<SalesOrder> => {
+  return axios.get(`${apiRoot}/sales/id/${id}`).then((res) => res.data);
+};
+
+export const completeOrderPrepSvc = (salesOrder: SalesOrder): Promise<any> => {
+  console.log('salesOrder HEREEEEEE', salesOrder);
```
Comment (danielonges) [inline src/services/salesService.ts:16]: remove this console log

**Review by danielonges — CHANGES_REQUESTED** [review]
need to refactor ah, if not we will accumulate tech debt as we go along

**Pull request closed without merging**